Core/src/main/java/com/plotsquared/core/setup/CommonSetupSteps.java

```diff
@@ -0,0 +1,288 @@
+            if (!isValidWorldName(argument)) {
+                MainUtil.sendMessage(plotPlayer, Captions.SETUP_WORLD_NAME_FORMAT + argument);
+                return this;
+            }
+            if (WorldUtil.IMP.isWorld(argument)) {
+                if (PlotSquared.get().hasPlotArea(argument)) {
+                    MainUtil.sendMessage(plotPlayer, Captions.SETUP_WORLD_NAME_TAKEN);
+                    return this;
+                }
+                MainUtil.sendMessage(plotPlayer, Captions.SETUP_WORLD_APPLY_PLOTSQUARED);
+            }
+            builder.worldName(argument);
+            plotPlayer.deleteMeta("setup");
+            String world;
+            if (builder.setupManager() == null) {
+                world = SetupUtils.manager.setupWorld(builder);
+            } else {
+                world = builder.setupManager().setupWorld(builder);
+            }
+            try {
+                plotPlayer.teleport(WorldUtil.IMP.getSpawn(world), TeleportCause.COMMAND);
+            } catch (Exception e) {
+                plotPlayer.sendMessage("&cAn error occurred. See console for more information");
+                e.printStackTrace();
+            }
+            sendMessage(plotPlayer, Captions.SETUP_FINISHED, builder.worldName());
+            return null;
+        }
+
+        @Nullable @Override public String getDefaultValue() {
+            return null;
+        }
+    };
+
+    @Getter @NotNull private final Collection<String> suggestions;
+    private final Caption description;
+
+    /**
+     *
+     * @param suggestions the input suggestions for this step
+     * @param description the caption describing this step
+     */
+    CommonSetupSteps(@NotNull Collection<String> suggestions, @NotNull Caption description) {
+        this.suggestions = suggestions;
+        this.description = description;
+    }
+
+    CommonSetupSteps(@NotNull Caption description) {
+        this.description = description;
+        this.suggestions = Collections.emptyList();
+    }
+
+    <E extends Enum<E>> CommonSetupSteps(@NotNull Class<E> argumentType, Caption description) {
+        this(enumToStrings(argumentType), description);
+    }
+
+    @Override public void announce(PlotPlayer<?> plotPlayer) {
+        MainUtil.sendMessage(plotPlayer, this.description);
+    }
+
+    private static <E extends Enum<E>> Collection<String> enumToStrings(Class<E> type) {
+        return Arrays.stream(type.getEnumConstants()).map(e -> e.toString().toLowerCase()).collect(Collectors.toList());
+    }
+
+    private static SettingsNodesWrapper wrap(String plotManager) {
+        return new SettingsNodesWrapper(SetupUtils.generators.get(plotManager).getPlotGenerator()
+                .getNewPlotArea("CheckingPlotSquaredGenerator", null, null, null)
+                .getSettingNodes(), CHOOSE_WORLD_NAME);
+    }
+
+    private static boolean isValidWorldName(String s) {
```

Comment:
Is that still necessary? World names can be uppercase as well. The commit forcing world names to be lowercase refers to 1.13 and has been reverted years ago.

Reply:
We can change that to match `[a-zA-Z0-9]` instead, yeah (or are there any other restrictions/possibilities?)

Reply:
I wanted to get a rid of it, since it's no longer necessary cb/b wise. Or do we rely on lowercased world names?

Reply:
There's no reason to allow uppercase world names imo

Reply:
We could add a configuration to allow world names that are not lowercase. I have been strongly opposed to allowing non lowercase due to legacy setups possibly being messed up. I also want to reduce the number of calls to `toLowerCase()` and `equalsIgnoreCase()`